### PR TITLE
JDBC sources: improve timestamptz handling

### PR DESCRIPTION
**Pull request by edgao**
I have no idea what the implications of this are. I believe that current behavior is to shift all timestamptz columns by our local offset (e.g. `2023-01-01 12:00Z` would be returned as `2023-01-01 04:00Z` if we're running in UTC-8).

WIthout this change: using a Redshift JDBC connection: `jdbcDatabase.queryJsons("SELECT CURRENT_TIMESTAMP, CAST(CURRENT_TIMESTAMP AS VARCHAR)` returns 
```json
{
  "timestamptz":"2023-11-16T15:23:58.503985Z",
      "varchar":"2023-11-16 23:23:58.503985+00"
}
```
. The varchar column is correct, because it does the timestamp->string conversion on the server side. The `timestamptz` column is _wrong_. It correctly swapped the time to my local time (UTC-8), but appended a `Z` suffix instead of `-08:00`.

This happens because redshift JDBC returns a ResultSet metadata, where `rs.getMetadata().getColumnType()` is 93 (i.e JDBCType.TIMESTAMP, i.e. without timezone). Fortunately, `rs.getMetadata().getColumnTypeName()` correctly returns `timestamptz`, so we can still recognize the _real_ type... but this feels like a massive hack.

Also add more handling in `putTimestampWithTimezone`, because `getObject(..., OffsetTime.class)` throws an exception.

### CODE REVIEW

File: airbyte-integrations/connectors/source-bing-ads/metadata.yaml [hunk 1]
```diff
@@ -25,11 +25,11 @@ data:
   name: Bing Ads
   registries:
     cloud:
-      dockerImageTag: 1.13.0  #https://airbytehq-team.slack.com/archives/C0662JB7XPU
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-bing-ads/metadata.yaml:28]: revert?

File: airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSourceOperations.java [hunk 2]
```diff
@@ -4,25 +4,62 @@
...
+    final JDBCType columnType = safeGetJdbcType(columnTypeInt);
+
+    if ("timestamptz".equalsIgnoreCase(resultSet.getMetaData().getColumnTypeName(colIndex))) {
+      // Massive hack. Sometimes the JDBCType is TIMESTAMP (i.e. without timezone)
+      // even though it _should_ be TIMESTAMP_WITH_TIMEZONE.
+      // Check for this case explicitly.
+      putTimestampWithTimezone(json, columnName, resultSet, colIndex);
+    } else {
+      // https://www.cis.upenn.edu/~bcpierce/courses/629/jdkdocs/guide/jdbc/getstart/mapping.doc.html
+      switch (columnType) {
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSourceOperations.java:43]: super.copyToJsonField? Seems much more accurate, no?

File: airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSourceOperations.java [hunk 3]
```diff
@@ -44,6 +81,19 @@ protected void setTimestamp(final PreparedStatement preparedStatement, final int
     preparedStatement.setTimestamp(parameterIndex, Timestamp.valueOf(date));
   }
 
+  @Override
+  protected void putTimestampWithTimezone(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index)
+      throws SQLException {
+    try {
+      final OffsetDateTime timestamptz = getObject(resultSet, index, OffsetDateTime.class);
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-redshift/src/main/java/io/airbyte/integrations/source/redshift/RedshiftSourceOperations.java:88]: super.putTimestampeWithTimezone?  Why duplicate code?

**Review by stephane-airbyte — APPROVED** [review]

**Comment by edgao** [comment]
redshift tests are failing exactly the same way master is failing (and master has been failing for multiple months). I added an entry to the changelog so we don't forget about this PR. approve-and-merging.

**Comment by octavia-approvington** [comment]
Send it! <br/> rocket socks

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**